functions/anonymous-functions.md

```diff
@@ -0,0 +1,198 @@
+
+## Syntax
+
+An anonymous function is a function created with any valid syntax, but it is not referenced by a variable.
+
+Instead, an **anonymous function** is a function defined in the one place that it's used.
+
+Technically, many of the methods in objects we created are anonymous functions!
+
+<!-- prettier-ignore-start -->
+```js
+const myTedTalk = {
+  title: 'Inspiration From Inspiring Things',
+  summary: 'A story about where inspiration comes from.',
+  summarize: function() {
+    return `${this.title}: ${this.summary}`;
+  }
+}
+
+myTedTalk.summarize();
+```
+<!-- prettier-ignore-end -->
+
+We never _named_ the value of the `summarize` property, or referenced it in a variable. We defined the function in the one place where it's needed, anonymously!
+
+## Uses for Anonymous Functions: `forEach` Loops
+
+An anonymous function is useful when create for-each loops.
+
+The `forEach` loop in JavaScript is a function that operates off of a collection (like an array). It takes in one argument: a **function**.
+
+_The passed-in function_ takes in one argument, which represents one item from the collection.
+
+<!-- prettier-ignore-start -->
+```js
+const myArray = ['red', 'yellow', 'blue'];
+
+myArray.forEach( item => {
+  console.log('Inside this anonymous function, each item of myArray will be accessed as item:', item);
+});
+```
+<!-- prettier-ignore-end -->
+
+| <div style="min-width:200px;"> Piece of Code </div> | Notes                                                                                                                                    |
+| --------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------- |
+| `const myArray = [...];`                            | Here we create an array, `myArray`.                                                                                                      |
+| `myArray.forEach(...);`                             | To iterate over each element in `myArray`, we call the `forEach` function off `myArray`.                                                 |
+| `item => { ... }`                                   | The `forEach` function takes in one argument: another function! This anonymous function is using arrow function syntax.                  |
+| `item`                                              | The anonymous function we pass into `forEach` should take in one argument itself. This argument will always represent the iterated item. |
+| `console.log(..., item);`                           | In the function body, we can access the value of `item` and read it!                                                                     |
+
+Anonymous functions can be arrow functions or non-arrow functions! Whether we use an arrow function or non-arrow function depends on the context of the problem, and which form of `this` we need.
+
```

Comment:
I think it's pretty safe to make the recommendation that arrow functions should generally be our go to for anonymous functions.

Add a callout that talks about some of the strangeness of `forEach` and point the reader at for/of as something to research. Editorial, but I think reasonable.

```suggestion

More often than not, our first instinct for creating anonymous functions should be to use arrow functions.

### !callout-info

## Many Ways to Iterate

JavaScript could be said to have _too many_ ways to iterate over arrays. In general, `forEach` is the _least flexible_ way of iterating over an array. As an example, notice that there is no way to stop iterating early if we were looking for a particular value in the array. `forEach` should only be used when we know that we need to apply the supplied function operations to _every_ item in the array.

<br />

Even then, one of JavaScript's several `for` loop styles, especially the `for`/`of` loop may give us iteration behavior more like we might be used to in other languages. Follow your curiosity!

### !end-callout

```